js/sparkswap.js

```diff
@@ -125,7 +126,39 @@ module.exports = class sparkswap extends Exchange {
     }
 
     async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
-        throw new ExchangeError ('Not Implemented');
+        await this.loadMarkets ();
+        let orderType = undefined;
+        if (side === 'buy') {
+            orderType = 'BID';
+        } else if (side === 'sell') {
+            orderType = 'ASK';
+        }
+        const order = {
+            'market': this.marketId (symbol),
+            'amount': amount.toString (),
+            'side': orderType,
+        };
+        const marketOrder = (type === 'market');
+        const limitOrder = (type === 'limit');
+        let timeInForceIsRequired = false;
+        if (limitOrder) {
+            if (price === undefined) {
+                throw new InvalidOrder (this.id + ' createOrder method requires a price argument for a ' + type + ' order');  
+            }
+            order['limit_price'] = price.toString ();
+            timeInForceIsRequired = true;
+        }
+        if (marketOrder) {
+            order['is_market_order'] = true;
+        }
+        if (timeInForceIsRequired) {
+            order['time_in_force'] = this.options['defaultTimeInForce']; // 'GTC' = Good To Cancel (default), 'IOC' = Immediate Or Cancel
```

Comment:
instead of having a `timeInForceIsRequired` bool, we can just default this value above